File: Backend/Infrastructure.MyShop/Models/DTO/ProductDTO/ProductCreateDTO.cs
using Microsoft.AspNetCore.Http;

namespace Infrastructure.MyShop.Models.DTO.ProductDTO
{
    public class ProductCreateDTO
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int Price { get; set; }
        public List<IFormFile> Images { get; set; }
        public int? CategoryId { get; set; }
    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/dto/productDTO/ProductCreateDTO.java
package org.example.Infrastructure.dto.productDTO;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

@Data
public class ProductCreateDTO {
    private String name;
    private String description;
    private double price;
    private List<MultipartFile> images;
    private int categoryId;
}